### ISSUE TITLE
Bug in FastAPI doc.

### ISSUE THREAD

**Issue by matveyDev** [tiangolo/fastapi#4684]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Header


app = FastAPI()


@app.get("/items/")
async def read_items(x_token: list[str] | None = Header(None)):
    return {"X-Token values": x_token}
```

### Description

Documentation says:
The response would be like:
{
    "X-Token values": [
        "bar",
        "foo"
    ]
}
--------------------------
BUT, I receive this:
{
  "X-Token values": [
    "1,2,3"
  ]
}

https://fastapi.tiangolo.com/tutorial/header-params/
![image](https://user-images.githubusercontent.com/99687428/158199593-f76ccb88-d5a4-4930-b0e2-67661d44cedc.png)
![image](https://user-images.githubusercontent.com/99687428/158199548-9f02b6b1-d9ab-4231-9969-83594d4c746a.png)

### Operating System

Linux

### Operating System Details

pop-os

### FastAPI Version

0.75.0

### Python Version

3.10.0rc2

**Comment by harunyasar** [comment]
Good spot! Apparently, `curl` is not being generated correctly.

**Comment by MarlonCorreia** [comment]
Yeah. But I think it's [Swagger UI](https://github.com/swagger-api/swagger-ui) fault. In applications such as postman the request works the same as the docs. I'm not sure what's recommended in those cases, maybe opening a issue in that project?

**Comment by matveyDev** [comment]
> 

Okay! Thank you for information.
Yep, maybe we should!
